cobbler/remote.py

```diff
@@ -2234,64 +2226,69 @@ def xapi_object_edit(
-            if interface is None:
-                # If the interface is not existing, create a new one.
-                interface = system.NetworkInterface(self.api)
-            for attribute_key in attributes:
-                if self.__is_interface_field(attribute_key):
-                    if hasattr(interface, attribute_key):
-                        setattr(interface, attribute_key, attributes[attribute_key])
-                    else:
-                        self.logger.warning(
-                            'Network interface field "%s" could not be set. Skipping it.',
-                            attribute_key,
-                        )
-                else:
-                    self.logger.debug(
-                        "Field %s was not an interface field.", attribute_key
-                    )
-            system_to_edit.interfaces.update({interface_name: interface})
-        elif "delete_interface" in attributes:
+        :param handle: XML-RPC handle for the system that is being edited.
+        :param attributes: The attributes that are being passed by the CLI to the server.
+        :param object_name: The system name.
+        """
+        if "delete_interface" in attributes:
             if attributes.get("interface") is None:
                 raise ValueError("Interface is required for deletion.")
             system_to_edit = self.__get_object(handle)
             system_to_edit.delete_interface(attributes.get("interface"))
-        elif "rename_interface" in attributes:
+            return
+
+        if "rename_interface" in attributes:
             system_to_edit = self.__get_object(handle)
             system_to_edit.rename_interface(
                 attributes.get("interface", ""), attributes.get("rename_interface", "")
             )
+            return
+
+        # This if is taking care of interface logic. The interfaces are a dict, thus when we get the obj via
+        # the api we get references to the original interfaces dict. Thus this trick saves us the pain of
+        # writing the modified obj back to the collection. Always remember that dicts are mutable.
+        system_to_edit = self.__get_object(handle)
+        if system_to_edit is None:
+            raise ValueError(
+                f'No system found with the specified name (name given: "{object_name}")!'
+            )
+
+        # If we don't have an explicit interface name use the default interface or require an explicit
+        # interface if default cannot be found.
+        if len(system_to_edit.interfaces) > 1 and attributes.get("interface") is None:
+            if "default" not in system_to_edit.interfaces.keys():
+                raise ValueError("Interface is required.")
+            interface_name = "default"
+        if len(system_to_edit.interfaces) == 1:
+            interface_name = attributes.get(
+                "interface", next(iter(system_to_edit.interfaces))
```

Comment:
Just a question if you don't mind. Over what are we iterating here? I just was noticing that the condition to be at this point is that `len(system_to_edit.interfaces) == 1`.

Reply:
We are iterating over the `NetworkInterface` dictionary here.

The reason we want to use this solution is because a Dict has no numeric index and we don't know the name of the interface.

Reply:
Thanks! I understand the reason now.